### PR TITLE
[master] Fix for handling of inline objects in content type for reusable request bodies

### PR DESCRIPTION
**Pull request by lnash94**
## Purpose
Fixes for two issues:

01. Fixes https://github.com/ballerina-platform/openapi-tools/issues/1484
02. This includes the fix for  **microsoft.dynamics365businesscentral** issue
Issue: **microsoft.dynamics365businesscentral** connector was assigning generate json value when it has the requestBody scenario similar to the above issue.
Reason: Given connector's inline object defined without the schema type object.

## Automation tests
 - Unit tests - Added
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-cli/src/test/resources/generators/schema/ballerina/schema_with_request_body_ref.bal [hunk 1]
```diff
@@ -14,10 +14,3 @@ public type Dog record {
     *Pet;
     boolean bark?;
 };
-
```
Comment (aneeshafedo) [inline openapi-cli/src/test/resources/generators/schema/ballerina/schema_with_request_body_ref.bal:17]: Why this `Pet` record is not generating now ? `Pet` is used in multiple places too

Comment (lnash94) [reply openapi-cli/src/test/resources/generators/schema/ballerina/schema_with_request_body_ref.bal]: It is a mistake by copying the generated code from resources  `Pet` is there. I updated the resource file

File: openapi-cli/src/test/resources/generators/client/ballerina/request_body_with_ref.bal [hunk 2]
```diff
@@ -38,7 +38,7 @@ public isolated client class Client {
     #
     # + payload - Return from creating a pet
     # + return - Successful operation
-    remote isolated function createPet(CreatedPet payload) returns http:Response|error {
+    remote isolated function createPet(CreatedPet_requestBody payload) returns http:Response|error {
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/resources/generators/client/ballerina/request_body_with_ref.bal:41]: Using title case for the suffix looks better to me. WDYT? @lnash94 
```suggestion
    remote isolated function createPet(CreatedPet_RequestBody payload) returns http:Response|error {
```

Comment (lnash94) [reply openapi-cli/src/test/resources/generators/client/ballerina/request_body_with_ref.bal:41]: +1, I will update with the title case in the suffix

**Review by aneeshafedo — APPROVED** [review]
LGTM

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**